app/src/components/map/DatasetPopup.tsx

```diff
@@ -46,21 +46,21 @@ const useStyles = makeStyles(() => ({
   }
 }));
 
-const DatasetPopup: React.FC<React.PropsWithChildren<{ submissionSpatialComponentId: number }>> = ({
-  submissionSpatialComponentId
+const DatasetPopup: React.FC<React.PropsWithChildren<{ submissionSpatialComponentIds: number[] }>> = ({
+  submissionSpatialComponentIds
 }) => {
   const classes = useStyles();
   const api = useApi();
   const history = useHistory();
 
   const dataLoader = useDataLoader(() => {
-    return api.search.getSpatialMetadata(submissionSpatialComponentId);
+    return api.search.getSpatialMetadata(submissionSpatialComponentIds);
   });
 
   dataLoader.load();
 
   const { isLoading, isReady } = dataLoader;
-  const data = dataLoader.data as BoundaryCentroidFeatureProperties;
+  const data = (dataLoader.data || []) as BoundaryCentroidFeatureProperties[];
```

Comment:
Lets chat about a potential way to improve this, so we don't need the cast.

Reply:
Added generic type to API hook, as per out conversation